hermes/probe/create/create_file.go

```diff
@@ -21,91 +27,99 @@ const (
 }
 
-func (r *randomHermesFileReader) readDone() bool {
+func (r *randomFileReader) readDone() bool {
 	return r.i >= r.size
 }
 
-func (r *randomHermesFileReader) bytesLeft() int {
+func (r *randomFileReader) bytesLeft() int {
 	return (r.size - r.i)
 }
 
-func (r *randomHermesFileReader) bufferTooLong(buffer []byte) bool {
-	return len(buffer) > r.bytesLeft()
+func (r *randomFileReader) bufferTooLong(buf []byte) bool {
+	return len(buf) > r.bytesLeft()
 }
 
-func (r *randomHermesFileReader) Read(buf []byte) (n int, err error) {
+func (r *randomFileReader) Read(buf []byte) (n int, err error) {
 	if r.readDone() {
 		return 0, io.EOF
 	}
 	b := buf
 	if r.bufferTooLong(b) {
 		b = buf[:r.bytesLeft()]
 	}
-	n, err = r.rand.Read(b) //n is the length  of the buffer
+	n, err = r.rand.Read(b) // n is the length  of the buffer
 	if err != nil {
-		return n, err // in this case n=0 and we return 0 as nil can't be returned as a type int argument
+		return n, err // in this case n=0
 	}
 	r.i += n
 	return n, err
 }
 
 // Warning: NewReader is not thread safe
-func (f *RandomHermesFile) NewReader() *randomHermesFileReader {
-	return &randomHermesFileReader{size: f.Size, rand: rand.New(rand.NewSource(f.ID))} // ID will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
+func (f *RandomFile) NewReader() *randomFileReader {
+	// ID will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
+	return &randomFileReader{
+		size: f.Size,
+		rand: rand.New(rand.NewSource(f.ID)),
+	}
+}
+
+func NewRandomFile(fileID int64, fileSize int) (*RandomFile, error) {
+	if fileID < begin || fileID > end {
+		return &RandomFile{}, fmt.Errorf("the file ID provided %v wasn't in the required range [%v, %v]", fileID, begin, end)
+	}
+	if fileSize > maxFileSizeBytes || fileSize <= 0 {
+		return &RandomFile{}, fmt.Errorf("invalid argument: RandomFile.Size = %d; want 0 < size <= %d", fileSize, maxFileSizeBytes)
+	}
+	return &RandomFile{ID: int64(fileID), Size: fileSize}, nil
 }
 
-func (f *RandomHermesFile) CheckSum() ([]byte, error) {
+func (f *RandomFile) CheckSum() ([]byte, error) {
 	r := f.NewReader()
 	h := sha1.New()
 	if _, err := io.Copy(h, r); err != nil {
 		return nil, fmt.Errorf("io.Copy: %v", err)
 	}
-	// returns checksum in hex notation
 	return h.Sum(nil), nil
 }
 
-func (f *RandomHermesFile) FileName() (string, error) {
-	if f.ID < begin || f.ID > end {
-		return "", fmt.Errorf("The file ID provided %v wasn't in the required range [1,50]", f.ID)
-	}
-	if f.Size > maxFileSizeBytes {
-		return "", fmt.Errorf("The file size provided %v bytes exceeded the limit %v bytes", f.Size, maxFileSizeBytes)
-	}
-	if f.Size <= 0 {
-		return "", fmt.Errorf("The file size provided %v is not a positive number as required", f.Size)
-	}
+func (f *RandomFile) FileName() (string, error) {
 	checksum, err := f.CheckSum()
 	if err != nil {
 		return "", err
```

Comment:
You should anote this error - I would be somewhat surprised if Filename() failed because of an error with `io.copy`

Also, does FileName need to be exported?

Reply:
It does not need to be exported changed to a non-exported function! checksum as well

Reply:
Errors wrapped with %w now

Reply:
apart from Fatalf because there %v is needed